packages/apputils/src/toolbar.tsx

```diff
@@ -244,6 +244,66 @@ export class Toolbar<T extends Widget = Widget> extends Widget {
     return true;
   }
 
+  /**
+   * Insert an item into the toolbar at the after a target item.
+   *
+   * @param at - The target item to insert after.
+   *
+   * @param index - The index at which to insert the item.
+   *
+   * @param name - The name of the item.
+   *
+   * @param widget - The widget to add.
+   *
+   * @returns Whether the item was added to the toolbar. Returns false if
+   *   an item of the same name is already in the toolbar.
+   *
+   * #### Notes
+   * The index will be clamped to the bounds of the items.
+   * The item can be removed from the toolbar by setting its parent to `null`.
+   */
+  insertAfter(at: string, name: string, widget: T): boolean {
+    return this._insertRelative(at, 1, name, widget);
+  }
+
+  /**
+   * Insert an item into the toolbar at the before a target item.
+   *
+   * @param at - The target item to insert before.
+   *
+   * @param index - The index at which to insert the item.
+   *
+   * @param name - The name of the item.
+   *
+   * @param widget - The widget to add.
+   *
+   * @returns Whether the item was added to the toolbar. Returns false if
+   *   an item of the same name is already in the toolbar.
+   *
+   * #### Notes
+   * The index will be clamped to the bounds of the items.
+   * The item can be removed from the toolbar by setting its parent to `null`.
+   */
+  insertBefore(at: string, name: string, widget: T): boolean {
+    return this._insertRelative(at, 0, name, widget);
+  }
+
+  private _insertRelative(
+    at: string,
+    offset: number,
```

Comment:
I was a bit worried that this would fail if the offset was greater than 0 or 1 and that brought it outside of the bounds of the toolbar. But that is handled in the `Layout` class, so we are good to go!

https://github.com/phosphorjs/phosphor/blob/849b2fbbf394d39147242ad1ec958e50d94ce110/packages/widgets/src/panellayout.ts#L104

Reply:
Yup, I wanted to build a more general solution if possible — the bounds are clamped.